src/ape/managers/compilers.py

```diff
@@ -96,6 +93,31 @@ def compile(
 
         return contract_types_dict  # type: ignore
 
+    def compile_project(
+        self, contract_filepaths: List[Path], project_path: Path, contracts_path: Path
+    ) -> Dict[str, ContractType]:
+        """
+        Temporarily change the project context and compile the project.
+        Args:
+            contract_filepaths (List[pathlib.Path]): The list of files to compile,
+              as ``pathlib.Path`` objects. **NOTE**: The source files must be
+              present in the given ``contracts_path`` directory.
+            project_path (pathlib.Path): The root project path.
+            contracts_path (pathlib.Path): The project contracts source directory.
+        Returns:
+            Dict[str, ``ContractType``]: A mapping of contract names to their type.
+        """
+        compilers = None
+        if "registered_compilers" in self.__dict__:
+            compilers = self.__dict__.pop("registered_compilers")
+
+        try:
+            with self.config.using_project(project_path, contracts_path):
+                return self.compile(contract_filepaths)
+        finally:
+            if compilers:
+                self.__dict__["registered_compilers"] = compilers
+
```

Comment:
This is kind of wild. What is the concern about the compilers being already registered as plugins for other projects?

I would think that the plugins shouldn't really have to change, e.g. if you are a Vyper-only project, but you use a dependency that is Solidity, then you should have to have both plugins installed (otherwise it doesn't compile the Solidity dependencies). But I guess I should understand more how this is working first.

Reply:
hmm, perhaps the config can be reloaded only when a new project is loaded, and then we also make sure that values like `import_remappings` are always using the config value

Reply:
Let me experiment with that sentiment.

Reply:
I was able to remove this method entirely!

Now, the `registered_compilers` is cached per project. So if you change project's, it will re-load.

Reply:
sick!